clientv3/lease.go

```diff
@@ -163,6 +170,13 @@ func (l *lessor) Revoke(ctx context.Context, id LeaseID) (*LeaseRevokeResponse,
 }
 
 func (l *lessor) KeepAlive(ctx context.Context, id LeaseID) (<-chan *LeaseKeepAliveResponse, error) {
+	l.rc.mu.Lock()
```

Comment:
I don't think this belongs here since the method delegates the client connection handling to the goroutine